### DISCUSSION TITLE
Should edges array be ordered

### DISCUSSION THREAD

**Discussion by Nirosh-cartrabbit** [xyflow/xyflow#3212] [Q&A]
Do we need to have a ordered edges or node array all the time ah. Cant i just push the a new node or edge to array or should i need to slice and find exact position and push the object ah??

The issue is i have a node where it has two source handle true and false when i add a new node below one of the edge of source handle means my edges and nodes get re-positioned initially it is left and right means it goes right and left when i add under false source handle edge means.

this is the reference image [URL](https://imgur.com/a/iYnfF9X)

Thank you in advance :100:

**Comment by OsamaKhalilMohamed** [comment]
i'm facing the same problem where i have a filterNode with two routes and at some situations the left branch swaps with the right one, do u have any update on this issue please?
